src/uu/env/src/env.rs

```diff
@@ -187,6 +187,7 @@ pub fn uu_app() -> Command {
                 .help("process and split S into separate arguments; used to pass multiple arguments on shebang lines")
         ).arg(
             Arg::new("argv0")
+                .overrides_with("argv0")
```

Comment:
Can we have a test for this?

I'm not sure I understand all the implications; can `--argv0` be overridden from within `-S`? How do we *want* it to behave in scenarios like `env -a name1 -vvS"-a name2" -a name3 ./myprog`? No need to fix it all in one fell swoop, but if we (unintentionally) support some specific behavior then we should also test for it.

Reply:
done. I added test cases for normal override and one for the mix with "-S".
But I think that the first group of tests would not have been strictly needed as this is a framework feature and thus its more or less safe to assume that it works.